tf6muxserver/mux_server_ValidateProviderConfig.go

```diff
@@ -42,16 +43,22 @@ func (s muxServer) ValidateProviderConfig(ctx context.Context, req *tfprotov6.Va
 			resp.Diagnostics = append(resp.Diagnostics, res.Diagnostics...)
 		}
 
-		if res.PreparedConfig != nil {
-			// This could check equality to bypass the error, however
-			// DynamicValue does not implement Equals() and previous mux server
-			// implementations have not requested the enhancement.
-			if resp.PreparedConfig != nil {
-				return nil, fmt.Errorf("got a ValidateProviderConfig PreparedConfig response from multiple servers, not sure which to use")
-			}
+		// Do not check equality on missing PreparedConfig or unset PreparedConfig
+		if res.PreparedConfig == nil {
+			continue
+		}
 
-			resp.PreparedConfig = res.PreparedConfig
+		equal, err := dynamicValueEquals(schemaType(s.providerSchema), res.PreparedConfig, resp.PreparedConfig)
+
+		if err != nil {
+			return nil, fmt.Errorf("unable to compare PrepareProviderConfig PreparedConfig responses: %w", err)
 		}
+
+		if !equal {
+			return nil, fmt.Errorf("got different PrepareProviderConfig PreparedConfig response from multiple servers, not sure which to use")
+		}
+
+		resp.PreparedConfig = res.PreparedConfig
```

Comment:
In this loop, at the first chance that we get back a `res` and `resp` is still `nil`, we assign `resp = res` (line 36 above).
Than later on we keep looping, and if we reach line 61, it means we found a `res.PrepareConfig` identical to `resp.PrepareConfig`, so we set `resp.PrepareConfig` again. Doesn't that risk creating a `resp` that is based on 2 difference "sources"?

Maybe that's precisely what you are after, and I just haven't learned enough of `mux` yet, but it feels... odd (for lack of better words).

I suspect this is the same above for protocol `5.0`.

Reply:
> Doesn't that risk creating a resp that is based on 2 difference "sources"?

Technically, yes, but the data is equal by this point, so it shouldn't matter. We could wrap this with `if resp.PreparedConfig == nil { resp.PreparedConfig = res.PreparedConfig }` conditional or prepend a `if resp.PreparedConfig != nil { continue }` before here, but both felt extraneous when the final result isn't changing. Willing to change it for clarity if it helps though.